test/server/cards/02_SHD/leaders/QiraIAloneSurvived.spec.ts

```diff
@@ -0,0 +1,75 @@
+describe('Qi\'ra, I Alone Survived', function() {
+    integration(function(contextRef) {
+        describe('Qi\'ra\'s leader undeployed ability', function() {
+            beforeEach(function() {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['death-star-stormtrooper', 'scout-bike-pursuer'],
+                        leader: 'qira#i-alone-survived',
+                        resources: 4
+                    },
+                    player2: {
+                        groundArena: ['battlefield-marine']
+                    },
+                });
+            });
+
+            it('should damage a friendly unit and give it a shield', function() {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.qira);
+                expect(context.player1).toBeAbleToSelectExactly([context.scoutBikePursuer, context.deathStarStormtrooper]);
+                context.player1.clickCard(context.scoutBikePursuer);
+
+                // check damage and shield are applied
+                expect(context.scoutBikePursuer.damage).toBe(2);
+                expect(context.scoutBikePursuer).toHaveExactUpgradeNames(['shield']);
+                expect(context.player2).toBeActivePlayer();
+            });
+
+            it('should damage (and kill) a friendly unit with 3 power or less', function() {
```

Comment:
This is definitely an instance where we could consolidate tests to reduce run time. Could we merge this one into the above by setting `context.qira.exhausted = false;` in between and just have a comment explaining the goal of each case. Unfortunately ugly but necessary for now